src/main/webapp/app/exercises/shared/participation-submission/participation-submission.component.ts

```diff
@@ -174,27 +179,7 @@ export class ParticipationSubmissionComponent implements OnInit {
     }
 
     getCommitUrl(submission: ProgrammingSubmission): string | undefined {
```

Comment:
Return value is `string`.
However, it might make sense to consider returning undefined instead of an empty string in the `createCommitUrl()` and handling the undefined event.

Reply:
I just extracted the function from this component, but we can change that.